editor/plugins/mesh_instance_3d_editor_plugin.cpp

```diff
@@ -608,6 +617,20 @@ MeshInstance3DEditor::MeshInstance3DEditor() {
 	debug_uv->set_custom_minimum_size(Size2(600, 600) * EDSCALE);
 	debug_uv->connect(SceneStringName(draw), callable_mp(this, &MeshInstance3DEditor::_debug_uv_draw));
 	debug_uv_dialog->add_child(debug_uv);
+
+	navigation_mesh_dialog = memnew(ConfirmationDialog);
+	navigation_mesh_dialog->set_title(TTR("Create NavigationMesh"));
+	navigation_mesh_dialog->set_ok_button_text(TTR("Create"));
+
+	VBoxContainer *navigation_mesh_dialog_vbc = memnew(VBoxContainer);
+	navigation_mesh_dialog->add_child(navigation_mesh_dialog_vbc);
+
+	Label *navigation_mesh_l = memnew(Label);
+	navigation_mesh_l->set_text(TTR("Before converting a rendering mesh to a navigation mesh please verify:\n\n- The mesh is two-dimensional.\n- The mesh has no surface overlap.\n- The mesh has no self-intersection.\n- The mesh surfaces have indices.\n\nIf the mesh does not fulfill these requirements the pathfinding will be broken."));
```

Comment:
```suggestion
	navigation_mesh_l->set_text(TTR("Before converting a rendering mesh to a navigation mesh, please verify:\n\n- The mesh is two-dimensional.\n- The mesh has no surface overlap.\n- The mesh has no self-intersection.\n- The mesh surfaces have indices.\n\nIf the mesh does not fulfill these requirements the pathfinding will be broken."));
```

Also, couldn't mesh indices be generated automatically if they are currently missing?

Reply:
Sure but that is the status quo how that option works. The conversion has already never worked with meshes that had no index array, this just makes it clear upfront.